core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/auth/ExternalAuthorizationRule.scala

```diff
@@ -35,19 +36,19 @@ class ExternalAuthorizationRule(val settings: ExternalAuthorizationRule.Settings
                                 override val impersonation: Impersonation)
   extends BaseAuthorizationRule {
 
-  private val userMatcher = PatternsMatcher.create[User.Id](settings.users.toSet)
-
   override val name: Rule.Name = ExternalAuthorizationRule.Name.name
 
-  override protected val groupsPermittedByRule: PermittedGroupIds = settings.permittedGroupsLogic.permittedGroupIds
+  override val groupsLogic: GroupsLogic = settings.groupsLogic
+
+  private val userMatcher = PatternsMatcher.create[User.Id](settings.users.toSet)
 
   override protected def loggedUserPreconditionCheck(user: LoggedUser): Either[Unit, Unit] = {
     Either.cond(userMatcher.`match`(user.id), (), ())
   }
 
   override protected def userGroups[B <: BlockContext](blockContext: B,
                                                        user: LoggedUser,
-                                                       permittedGroupIds: Set[GroupIdLike])
+                                                       groupsPotentiallyPermittedByRule: GroupsPotentiallyPermittedByRule)
```

Comment:
I see that the `GroupsPotentiallyPermittedByRule' escapes the `BaseGroupsRule' context. That's suspicious. Together with the IllegalStateException exception thrown in the `LdapAuthorizationRule` class, makes me wonder if there is a problem here.

Reply:
I now removed completely the `GroupsPotentiallyPermittedByRule`. All necessary information is already present inside `GroupsLogic`. There was no exception thrown in previous iteration, I changed it after dicsussions. Now changed back to implementation without exceptions (correct situations guaranteed at the type level of rule config)